code/modules/mob/living/simple_animal/hostile/mining/goliath.dm

```diff
@@ -137,10 +149,37 @@
 					new /obj/effect/temp_visual/goliath_tentacle(t, src)
 			else
 				cached_tentacle_turfs -= t
+	if(target && prob(20) && juvgoliathsspawned <= 2) // Spawn juvenile goliaths to aid in combat. Maximum of 3.
+		var/mob/living/simple_animal/hostile/asteroid/goliathjuv/G = new /mob/living/simple_animal/hostile/asteroid/goliathjuv(loc)
+		G.admin_spawned = admin_spawned
+		G.GiveTarget(target)
+		G.friends = friends
+		G.faction = faction.Copy()
+		visible_message("<span class='warning'>[src] matter splits off to form a smaller monster!</span>")
+		juvgoliathsspawned++
+		G.creator = src
+	if(stat != DEAD && juvgoliathsspawned <= 4) // Pickup stray juveniles for a max of 5, up from 3 it can spawn
+		for(var/mob/living/simple_animal/hostile/asteroid/goliathjuv/G in GLOB.living_mob_list)
```

Comment:
Looping through every goliath in the game every life tick seems like it could get expensive, is there no better way to do this?